### PR TITLE
Adding ip5306

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Adding ip5306

## Motivation and Context

- Adding ip5306 using as power management for M5Stack

## How Has This Been Tested?

On a real device.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: devices/Ip5306/Ip5306.cs [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+        /// </summary>
+        /// <remarks>Typical valut is between 50 and 3150 milli Ampere. Values are capted to 50 for anything under or 3150 for anything higher.</remarks>
+        public ElectricCurrent ChargingCurrent
+        {
+            get
+            {
+                var buf = I2cRead(Register.CHG_DIG_CTL0) & 0b0001_1111;
+                // I=0.05+b0*0.1+b1*0.2+b2*0.4+b3*0.8+b4*1.6A
+                // I = buff * 100 + 50 mA
+                return ElectricCurrent.FromMilliamperes(buf * 100 + 50);
```
Comment (CarlosSardo) [inline devices/Ip5306/Ip5306.cs:321]: /nit: declare the 100 and 50 in constants with a meaningfull name, to make it easier to understand what these values mean.

File: devices/Ip5306/Ip5306.cs [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+
+            set
+            {
+                // Cap the values to 0 or 3100 mA
+                var currentmA = value.Milliamperes - 50;
+                if (currentmA < 0)
+                {
+                    currentmA = 0;
+                }
+                else if (currentmA > 3100)
```
Comment (CarlosSardo) [inline devices/Ip5306/Ip5306.cs:332]: /DRY: declare 3100 once, in a constant.

**Review by CarlosSardo — APPROVED** [review]

File: devices/Ip5306/Ip5306.cs (hunk 1 above)
Comment (Ellerbach) [reply devices/Ip5306/Ip5306.cs:321]: when used only once with a comment, it's ok to keep them like this.

File: devices/Ip5306/Ip5306.cs (hunk 2 above)
Comment (Ellerbach) [reply devices/Ip5306/Ip5306.cs:332]: same here, otherwise, you're right, it would be better in a constant if used multiple times

**Review by josesimoes — APPROVED** [review]
Awesome! Another one bites the dust. 💪🏻

**Pull request merged**